frontend/public/components/edit-yaml.jsx

```diff
@@ -303,40 +314,95 @@ export const EditYAML = connect(stateToProps)(
+            {`${create ? 'Create' : 'Edit'} ${_.get(model, 'label', kind)}`}
+          </div>}
+          <div className="co-p-has-sidebar">
+            <div className="co-p-has-sidebar__body">
+              <div className="yaml-editor" ref={r => this.editor = r} style={{height: this.state.height}}>
+                <div className="absolute-zero">
+                  <div className="full-width-and-height yaml-editor__flexbox">
+                    <div id={this.id} key={this.id} className="yaml-editor__acebox" />
+                    <div className="yaml-editor__buttons">
+                      {error && <p className="alert alert-danger"><span className="pficon pficon-error-circle-o"></span>{error}</p>}
+                      {success && <p className="alert alert-success"><span className="pficon pficon-ok"></span>{success}</p>}
+                      {stale && <p className="alert alert-info">
+                        <span className="pficon pficon-info"></span>This object has been updated. Click reload to see the new version.
+                      </p>}
+                      {create && <button type="submit" className="btn btn-primary" id="save-changes" onClick={() => this.save()}>Create</button>}
+                      {!create && !readOnly && <button type="submit" className="btn btn-primary" id="save-changes" onClick={() => this.save()}>Save</button>}
+                      {!create && <button type="submit" className="btn btn-default" id="reload-object" onClick={() => this.reload()}>Reload</button>}
+                      <button className="btn btn-default" id="cancel" onClick={() => this.onCancel()}>Cancel</button>
+                      {download && <button type="submit" className="btn btn-default pull-right hidden-sm hidden-xs" onClick={() => this.download()}><i className="fa fa-download"></i>&nbsp;Download</button>}
+                    </div>
                   </div>
                 </div>
               </div>
             </div>
+            <ResourceSidebar isCreateMode={create} kindObj={model} height={this.state.height} loadSampleYaml={this.loadSampleYaml_} downloadSampleYaml={this.downloadSampleYaml_} />
           </div>
-          <ResourceSidebar isCreateMode={create} kindObj={model} height={this.state.height} loadSampleYaml={this.loadSampleYaml_} downloadSampleYaml={this.downloadSampleYaml_} />
         </div>
       </div>;
+
+      return _.isFunction(connectDropTarget) ? connectDropTarget(editYamlComponent) : editYamlComponent;
     }
   }
 );
+
+const boxTarget = {
+  drop(props, monitor) {
+    if (props.onDrop && monitor.isOver()) {
+      props.onDrop(props, monitor);
+    }
+  },
+};
+
+export const EditYAMLComponent = DropTarget(NativeTypes.FILE, boxTarget, (connectObj, monitor) => ({
+  connectDropTarget: connectObj.dropTarget(),
+  isOver: monitor.isOver(),
+  canDrop: monitor.canDrop(),
+}))(EditYAML);
+
+
+export const DroppableEditYAML = withDragDropContext(class DroppableEditYAML extends React.Component {
```

Comment:
We should add `DroppableEditYAMLProps` and `DroppableEditYAMLState` to the `DroppableEditYAML` class